### PR TITLE
cli: Containerbuild clean

### PR DESCRIPTION
**Pull request by kalikiana**
Add a unit test (and fix) for `SNAPCRAFT_CONTAINER_BUILDS=1 clean`.

### CODE REVIEW

**Comment by elopio** [comment]
This looks correct to me. However, I'm having problems to test it.

I'm failing to build the snap with this error: http://paste.ubuntu.com/24930727/

Then the container is stopped. When I run clean, the first thing it tries to do is to start it, and it fails.

@kalikiana do you see something wrong in my first snapcraft call? And in the second, wouldn't it be better not even trying to start the container? just remove it?

File: snapcraft/internal/lxd.py [hunk 1]
```diff
@@ -219,7 +219,7 @@ def _finish(self):
 
     def execute(self, step='snap', args=None):
         super().execute(step, args)
-        if step == 'clean' and not args:
+        if step == 'clean' and 'pull' in args:
```
Comment (elopio) [inline snapcraft/internal/lxd.py]: Don't we have to check here if there are no parts in `args`? Maybe, this clean command is to clean only one part, and leave the rest.

**Comment by kalikiana** [comment]
@elopio I have no idea what your error is about. I suggest you open a forum topic for investigation. This branch isn't changing the implementation, only the passing of arguments.
I don't think we can "just" remove the container because then we wouldn't be cleaning anything in the project folder. But, this would also be a nice forum topic. We can maybe find ways to optimize by cleaning outside of the container (and investigate potential problems with that).

**Comment by elopio** [comment]
@kalikiana there's clearly a bug, not introduced by this branch but it prevents me from testing this branch. I reported it here, with simple steps to reproduce: https://bugs.launchpad.net/snapcraft/+bug/1700703

It happens on all my machines, but I made sure to reproduce it in a clean environment. I assigned it to you. Please take a look.

**Pull request merged**